catalog/transformer/catalog-transformer-html/src/main/java/ddf/catalog/transformer/html/HtmlMetacardUtility.java

```diff
@@ -0,0 +1,157 @@
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HtmlMetacardUtility {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HtmlMetacardTransformer.class);
+
+  private TemplateLoader templateLoader;
+
+  private Handlebars handlebars;
+
+  private ValueResolver[] resolvers;
+
+  private Template template;
+
+  private MimeType mimeType;
+
+  private static final String TEMPLATE_DIRECTORY = "/templates";
+
+  private static final String TEMPLATE_SUFFIX = ".hbs";
+
+  private static final String HTML_TEMPLATE = "template";
+
+  private List<HtmlExportCategory> categoryList;
+
+  public HtmlMetacardUtility() {
+    try {
+      this.mimeType = new MimeType("text/html");
+    } catch (MimeTypeParseException e) {
+      LOGGER.warn("Failed to apply mimetype text/html");
+    }
+
+    this.templateLoader = new ClassPathTemplateLoader();
+    this.templateLoader.setPrefix(TEMPLATE_DIRECTORY);
+    this.templateLoader.setSuffix(TEMPLATE_SUFFIX);
+
+    this.handlebars = new Handlebars(this.templateLoader);
+
+    this.resolvers =
+        new ValueResolver[] {
+          FieldValueResolver.INSTANCE, MapValueResolver.INSTANCE, JavaBeanValueResolver.INSTANCE
+        };
+
+    this.registerHelpers();
+
+    try {
+      this.template = this.handlebars.compile(HTML_TEMPLATE);
+    } catch (IOException e) {
+      LOGGER.warn("Failed to compile handlebars template {}", HTML_TEMPLATE, e);
+    }
+  }
+
+  public HtmlMetacardUtility(List<HtmlExportCategory> categoryList) {
+    this();
+    this.categoryList = categoryList;
+  }
+
+  private static List<HtmlExportCategory> sortCategoryList(List<HtmlExportCategory> categoryList) {
+    return categoryList
+        .stream()
+        .sorted(Comparator.comparing(HtmlExportCategory::getTitle))
+        .collect(Collectors.toList());
+  }
+
+  private void registerHelpers() {
+    handlebars.registerHelper(
+        "isBasicValue",
+        new IfHelper() {
+          public CharSequence apply(Object context, Options options) throws IOException {
+            return (context instanceof HtmlBasicValueModel) ? options.fn() : options.inverse();
+          }
+        });
+
+    handlebars.registerHelper(
+        "isEmptyValue",
+        new IfHelper() {
+          @Override
+          public CharSequence apply(Object context, Options options) throws IOException {
+            return (context instanceof HtmlEmptyValueModel) ? options.fn() : options.inverse();
+          }
+        });
+
+    handlebars.registerHelper(
+        "isMediaValue",
+        new IfHelper() {
+          @Override
+          public CharSequence apply(Object context, Options options) throws IOException {
+            return (context instanceof HtmlMediaModel) ? options.fn() : options.inverse();
+          }
+        });
+  }
+
+  public String buildHtml(List<HtmlMetacardModel> metacardModels) {
```

Comment:
Yes because it relies on the instance variable `template` and there's not a reason to call this method before/without instantiating an object of `HtmlMetacardUtility`